lib/hx509/revoke.c

```diff
@@ -628,41 +628,15 @@ load_crl(hx509_context context, const char *path, time_t *t, CRLCertificateList
     return ret;
 }
 
-/**
- * Add a CRL file to the revokation context.
- *
- * @param context hx509 context
- * @param ctx hx509 revokation context
- * @param path path to file that is going to be added to the context.
- *
- * @return An hx509 error code, see hx509_get_error_string().
- *
- * @ingroup hx509_revoke
- */
-
 HX509_LIB_FUNCTION int HX509_LIB_CALL
-hx509_revoke_add_crl(hx509_context context,
+_hx509_revoke_add_crl(hx509_context context,
 		     hx509_revoke_ctx ctx,
 		     const char *path)
 {
     void *data;
     size_t i;
     int ret;
 
-    if (strncmp(path, "FILE:", 5) != 0) {
-	hx509_set_error_string(context, 0, HX509_UNSUPPORTED_OPERATION,
-			       "unsupported type in %s", path);
-	return HX509_UNSUPPORTED_OPERATION;
-    }
-
-
-    path += 5;
-
-    for (i = 0; i < ctx->crls.len; i++) {
-	if (strcmp(ctx->crls.val[i].path, path) == 0)
-	    return 0;
-    }
```

Comment:
The above block prevents the inclusion of duplicate entries in `crls`.  Why should duplicate entries be permitted?

Reply:
They shouldn't and I added back the relevant code block.